applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/line_search_contact_strategy.h

```diff
@@ -212,83 +212,91 @@ class LineSearchContactStrategy :
     ///@{
-    
+
     bool mRecalculateFactor;         // To check if we recalculate or not the scale factor
 
     ///@}
     ///@name Protected Operators
     ///@{
-    
+
     /**
-     * Performs all the required operations that should be done (for each step) 
+     * Performs all the required operations that should be done (for each step)
      * before solving the solution step.
      * A member variable should be used as a flag to make sure this function is called only once per step.
      */
-        
+
     void InitializeSolutionStep() override
     {
         BaseType::InitializeSolutionStep();
-        
+
         // TODO: Add something if necessary
     }
-    
+
     /**
      * Here the database is updated
      */
-     
+
     void UpdateDatabase(
         TSystemMatrixType& A,
         TSystemVectorType& Dx,
         TSystemVectorType& b,
-        const bool MoveMesh
+        const bool MoveMesh,
+        const bool complete_update = false
     ) override
     {
-        typename TSchemeType::Pointer pScheme = this->GetScheme();
-        typename TBuilderAndSolverType::Pointer pBuilderAndSolver = this->GetBuilderAndSolver(); // FIXME: Separate in the parts of LM and displacement
-
-        TSystemVectorType aux(b.size()); //TODO: do it by using the space
-        TSparseSpace::Assign(aux, 0.5, Dx);
-
-        TSystemVectorType DxDisp(b.size()); 
-        TSystemVectorType DxLM(b.size()); 
-        ComputeSplitDx(Dx, DxDisp, DxLM);
-        
-        // Compute residual without update
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double roDisp;
-        double roLM;
-        ComputeMixedResidual(b, roDisp, roLM);
-        
-        // Compute half step residual
-        NRBaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rhDisp;
-        double rhLM;
-        ComputeMixedResidual(b, rhDisp, rhLM);
-
-        // Compute full step residual (add another half Dx to the previous half)
-        NRBaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rfDisp;
-        double rfLM;
-        ComputeMixedResidual(b, rfDisp, rfLM);
-
-        // We compute the parabola        
-        double XminDisp = 1e-3;
-        double XmaxDisp = 1.0;
-        double XminLM = 1e-3;
-        double XmaxLM = 1.0;
-        
-        ComputeParabola(XminDisp, XmaxDisp, rfDisp, roDisp, rhDisp);
-        ComputeParabola(XminLM, XmaxLM, rfLM, roLM, rhLM);
-        
-        // Perform final update
-        TSparseSpace::Assign(aux,-(1.0 - XmaxDisp), DxDisp);
-        TSparseSpace::UnaliasedAdd(aux,-(1.0 - XmaxLM), DxLM);
-        NRBaseType::UpdateDatabase(A,aux,b,MoveMesh);
+        if(complete_update == true)
```

Comment:
```suggestion
        if(CompleteUpdate == true)
```

BTW, this is not following the style of the brackets